### PR TITLE
fix oop jit bugs with reclaimed property records

### PR DESCRIPTION
**Pull request by MikeHolman**
when property records were reclaimed we should remove them from list of pending jit property records.
when reclaimed property later has pid reused, we should first delete the old record on the server map before adding new one.

### CODE REVIEW

File: lib/Runtime/Base/ThreadContext.cpp [hunk 1]
```diff
@@ -1277,9 +1277,10 @@ bool ThreadContext::IsActivePropertyId(Js::PropertyId pid)
 void ThreadContext::InvalidatePropertyRecord(const Js::PropertyRecord * propertyRecord)
 {
     InternalInvalidateProtoTypePropertyCaches(propertyRecord->GetPropertyId());     // use the internal version so we don't check for active property id
-
+#if ENABLE_NATIVE_CODEGEN
+    m_pendingJITProperties->Remove(propertyRecord);
```
Comment (leirocks) [inline lib/Runtime/Base/ThreadContext.cpp]: Need to update the server that the property record is removed? maybe maintain a removed list of propIds, do incremental removing similar to the incremental adding?

Comment (MikeHolman) [inline lib/Runtime/Base/ThreadContext.cpp]: This was something I considered doing, but it seemed like a memory optimization (reclaiming in the runtime keeps the size bounded: on server we'll never hold more than the highest propertyid number of records), and I was a bit worried about increasing complexity/perf cost.

That said, upon reflection I think you're right. It's not much more effort, and it gives us more complete story for property records. I'll add the removal support in tomorrow morning.

Comment (MikeHolman) [inline lib/Runtime/Base/ThreadContext.cpp]: @leirocks I've added support for JIT reclaiming property records

**Comment by leirocks** [comment]
:+1:

**Pull request merged**